core/src/main/scala/io/finch/Decode.scala

```diff
@@ -40,3 +41,36 @@ object Decode {
    */
   @inline def apply[A, CT <: String](implicit d: Aux[A, CT]): Aux[A, CT] = d
 }
+
+trait AdaptableDecode[A, CT] {
+
+  def apply(contentType: Option[String]): Decode.Aux[A, CT]
+
+}
+
+object AdaptableDecode {
+
+  def instance[A, CT](fn: Option[String] => Decode.Aux[A, CT]):  AdaptableDecode[A, CT] = new AdaptableDecode[A, CT] {
+    def apply(contentType: Option[String]): Decode.Aux[A, CT] = fn(contentType)
+  }
+
+  implicit def mkCnil[A, CTH <: String](implicit decode: Decode.Aux[A, CTH]): AdaptableDecode[A, CTH :+: CNil] =
+    instance(_ => decode.asInstanceOf[Decode.Aux[A, CTH :+: CNil]])
```

Comment:
I like that this preserves the behavior, but maybe in future, we'd need to change this and make sure to check if the given content-type matches.

Reply:
Same thing about `mkSingle`.